src/Sylius/Bundle/CoreBundle/EventListener/LockerListener.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CoreBundle\EventListener;
+
+use Doctrine\DBAL\LockMode;
+use Doctrine\ORM\EntityManagerInterface;
+use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Core\Model\ProductVariantInterface;
+use Sylius\Component\Product\Resolver\ProductVariantResolverInterface;
+use Sylius\Component\Resource\Model\VersionedInterface;
+use Symfony\Component\EventDispatcher\GenericEvent;
+use Webmozart\Assert\Assert;
+
+/**
+ * @author Grzegorz Sadowski <grzegorz.sadowski@lakion.com>
+ */
+final class LockerListener
```

Comment:
Then we should have a `LockerListener` that will have a simple logic for every object that uses `VersionedInterface` (use it for variant) and a `SimpleProductLockerListener` that will have a logic for locking simple product and register them on a corresponding events.

Is `Locker` a proper prefix? Shouldn't it be `Locking` or some other word? /cc @TheMadeleine